core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -461,15 +463,25 @@ private[log] class Cleaner(val id: Int,
         ByteBufferMessageSet.writeMessage(buffer, messageOffset.message, messageOffset.offset)
       MessageSet.messageSetSize(messagesIterable)
     } else {
+      val messageSetTimestamp = MessageSet.validateMagicValuesAndGetTimestamp(messages.map(_.message))
+      val firstAbsoluteOffset = messages.head.offset
       var offset = -1L
+      val magicValue = messages.head.message.magic
```

Comment:
We should use the magic configured for this topic and convert each message to the right version if needed, right?